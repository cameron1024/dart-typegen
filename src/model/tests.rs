use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
