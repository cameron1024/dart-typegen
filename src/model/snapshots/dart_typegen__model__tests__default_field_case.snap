---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    defaults: Some(
        Defaults {
            generate_to_string: None,
            generate_equals: None,
            union: None,
            field: Some(
                FieldOptions {
                    json_key_case: Some(
                        SpannedScalar {
                            value: Camel,
                            span: SourceSpan {
                                offset: SourceOffset(
                                    91,
                                ),
                                length: 7,
                            },
                        },
                    ),
                },
            ),
        },
    ),
    enums: [],
    classes: [
        Class {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Foo",
                span: SourceSpan {
                    offset: SourceOffset(
                        112,
                    ),
                    length: 5,
                },
            },
            fields: [
                Field {
                    name: SpannedScalar {
                        value: "multipleWords",
                        span: SourceSpan {
                            offset: SourceOffset(
                                128,
                            ),
                            length: 15,
                        },
                    },
                    ty: SpannedScalar {
                        value: "String",
                        span: SourceSpan {
                            offset: SourceOffset(
                                149,
                            ),
                            length: 8,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: None,
                    docs: None,
                    json_key: None,
                },
                Field {
                    name: SpannedScalar {
                        value: "evenMoreMultipleWords",
                        span: SourceSpan {
                            offset: SourceOffset(
                                166,
                            ),
                            length: 23,
                        },
                    },
                    ty: SpannedScalar {
                        value: "String",
                        span: SourceSpan {
                            offset: SourceOffset(
                                195,
                            ),
                            length: 8,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: None,
                    docs: None,
                    json_key: None,
                },
            ],
            docs: None,
            json_key_case: None,
            json_discriminant_value: None,
            extra_dart: [],
        },
    ],
    unions: [],
}
